File: app/src/androidTest/java/de/westnordost/streetcomplete/data/ApplicationDbTestCase.java
package de.westnordost.streetcomplete.data;

import android.database.sqlite.SQLiteOpenHelper;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import de.westnordost.streetcomplete.util.KryoSerializer;
import de.westnordost.streetcomplete.util.Serializer;

import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
import static org.junit.Assert.assertNotNull;

public class ApplicationDbTestCase
{
	private final static String DATABASE_NAME = "streetcomplete_test.db";

	protected SQLiteOpenHelper dbHelper;
	protected Serializer serializer;

	@Before public void setUpHelper()
	{
		serializer = new KryoSerializer();
		dbHelper = DbModule.sqliteOpenHelper(getInstrumentation().getTargetContext(), DATABASE_NAME);
	}

	@Test public void databaseAvailable()
	{
		assertNotNull(dbHelper.getReadableDatabase());
	}

	@After public void tearDownHelper()
	{
		dbHelper.close();
		getInstrumentation().getTargetContext().deleteDatabase(DATABASE_NAME);
	}
}

File: app/src/androidTest/java/de/westnordost/streetcomplete/data/ApplicationDbTestCase.kt
package de.westnordost.streetcomplete.data

import android.database.sqlite.SQLiteOpenHelper
import androidx.test.platform.app.InstrumentationRegistry
import kotlin.test.AfterTest
import kotlin.test.BeforeTest
import kotlin.test.Test
import kotlin.test.assertNotNull

open class ApplicationDbTestCase {
    protected lateinit var dbHelper: SQLiteOpenHelper
    protected lateinit var database: Database

    @BeforeTest fun setUpHelper() {
        dbHelper = StreetCompleteSQLiteOpenHelper(
            InstrumentationRegistry.getInstrumentation().targetContext,
            DATABASE_NAME
        )
        database = AndroidDatabase(dbHelper)
    }

    @Test fun databaseAvailable() {
        assertNotNull(dbHelper.readableDatabase)
    }

    @AfterTest fun tearDownHelper() {
        dbHelper.close()
        InstrumentationRegistry.getInstrumentation().targetContext
            .deleteDatabase(DATABASE_NAME)
    }

    companion object {
        private const val DATABASE_NAME = "streetcomplete_test.db"
    }
}
